pandas/core/frame.py

```diff
@@ -3967,8 +3967,15 @@ def _set_value(
         """
         try:
             if takeable:
-                series = self._ixs(col, axis=1)
-                series._set_value(index, value, takeable=True)
+                if isinstance(self._mgr, ArrayManager):
```

Comment:
can you not do this in internals? hate leaking ArrayManager semantics here

Reply:
What I added here is actually to call into the internals to do it, instead of the current frame-level methods. 
It's only possible for ArrayManager, though (see https://github.com/pandas-dev/pandas/pull/41878/files#r757450838 just below), which is the reason I added a check.